spec/support/shared_examples/block_matcher.rb

```diff
@@ -0,0 +1,102 @@
+RSpec.shared_examples "an RSpec block-only matcher" do
+  # Note: do not use `matcher` in 2 expectation expressions in a single
+  # example here. In some cases (such as `change { }.to(2)`), it will fail
+  # because using it a second time will apply `x += 2` twice, changing
+  # the value to 4.
+
+  matcher :always_passes do
+    supports_block_expectations
+    match do |actual|
+      actual.call
+      true
+    end
+  end
+
+  matcher :always_fails do
+    supports_block_expectations
+    match do |actual|
+      actual.call
+      false
+    end
+  end
+
+  def valid_expectation
+    expect { |block| valid_block(&block) }
+  end
+
+  def invalid_expectation
+    expect { |block| invalid_block(&block) }
+  end
+
+  include_examples "an RSpec matcher"
+
+  it 'preserves the symmetric property of `==`' do
+    expect(matcher).to eq(matcher)
+    # TODO: the following expectations fail due to
```

Comment:
@JonRowe I have two open questions on how to deal with the following examples.

Apart from this one and the pair below everything is green.